ui/src/openReadOnlyContent.ts

```diff
@@ -29,13 +33,33 @@ export async function openReadOnlyJson(node: { label: string, fullId: string },
     await openReadOnlyContent(node, content, '.json');
 }
 
-export async function openReadOnlyContent(node: { label: string, fullId: string }, content: string, fileExtension: string): Promise<void> {
+export async function openReadOnlyContent(node: INode, content: string, fileExtension: string, options?: TextDocumentShowOptions): Promise<ReadOnlyContent> {
     if (!contentProvider) {
         contentProvider = new ReadOnlyContentProvider();
         ext.context.subscriptions.push(workspace.registerTextDocumentContentProvider(scheme, contentProvider));
     }
 
-    await contentProvider.openReadOnlyContent(node, content, fileExtension);
+    return await contentProvider.openReadOnlyContent(node, content, fileExtension, options);
+}
+
+export class ReadOnlyContent {
+    private _contentProvider: ReadOnlyContentProvider;
+    private _node: INode;
+    private _fileExtension: string;
+
+    constructor(_contentProvider: ReadOnlyContentProvider, node: INode, fileExtension: string) {
+        this._contentProvider = _contentProvider;
+        this._node = node;
+        this._fileExtension = fileExtension;
+    }
+
+    public async append(content: string, options?: TextDocumentShowOptions): Promise<void> {
+        await this._contentProvider.openReadOnlyContent(this._node, content, this._fileExtension, { append: true, ...options });
+    }
+
+    public clear(): void {
+        this._contentProvider.clearContent(this._node, this._fileExtension);
```

Comment:
It's a bit odd to have `ReadOnlyContentProvider` and `ReadOnlyContent` referencing each other. Instead, change `ReadOnlyContentProvider._contentMap` to `Map<string, ReadOnlyContent>` and store the content directly in the `ReadOnlyContent` class